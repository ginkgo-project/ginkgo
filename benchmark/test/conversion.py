#!/usr/bin/env python3
import test_framework

# check that all input modes work:
# parameter
test_framework.compare_output(
    ["-input", '[{"size": 100, "stencil": "7pt"}]', "-formats", "coo,csr"],
    expected_stdout="conversion.simple.stdout",
    expected_stderr="conversion.simple.stderr",
)

# stdin
test_framework.compare_output(
    ["-formats", "coo,csr"],
    expected_stdout="conversion.simple.stdout",
    expected_stderr="conversion.simple.stderr",
    stdin='[{"size": 100, "stencil": "7pt"}]',
)

# input file
test_framework.compare_output(
    [
        "-input",
        str(test_framework.sourcepath / "input.mtx.json"),
        "-formats",
        "coo,csr",
    ],
    expected_stdout="conversion.simple.stdout",
    expected_stderr="conversion.simple.stderr",
)

# check that all conversions work
test_framework.compare_output(
    [
        "-input",
        '[{"size": 100, "stencil": "7pt"}]',
        "-formats",
        "coo,csr,ell,sellp,hybrid",
    ],
    expected_stdout="conversion.all.stdout",
    expected_stderr="conversion.all.stderr",
)

# profiler annotations
test_framework.compare_output(
    [
        "-input",
        '[{"size": 100, "stencil": "7pt"}]',
        "-formats",
        "coo,csr",
        "-profile",
        "-profiler_hook",
        "debug",
    ],
    expected_stdout="conversion.profile.stdout",
    expected_stderr="conversion.profile.stderr",
)
