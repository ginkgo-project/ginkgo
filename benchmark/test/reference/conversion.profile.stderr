Running on reference(0)
Running with 0 warm iterations and 1 running iterations
The random seed for right hand sides is 42
The formats are coo,csr
Running test case stencil(100, 7pt)
Matrix is of size (125, 125), 725
DEBUG: begin components::aos_to_soa
DEBUG: end   components::aos_to_soa
DEBUG: begin stencil(100, 7pt)
	Running conversion: coo-read
DEBUG: begin coo-read
DEBUG: begin repetition
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin copy
DEBUG: end   copy
DEBUG: end   repetition
DEBUG: end   coo-read
	Running conversion: coo-csr
DEBUG: begin coo-csr
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin repetition
DEBUG: begin copy(<typename>)
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin components::convert_idxs_to_ptrs
DEBUG: end   components::convert_idxs_to_ptrs
DEBUG: end   copy(<typename>)
DEBUG: end   repetition
DEBUG: end   coo-csr
	Running conversion: csr-read
DEBUG: begin csr-read
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin repetition
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin components::convert_idxs_to_ptrs
DEBUG: end   components::convert_idxs_to_ptrs
DEBUG: end   repetition
DEBUG: end   csr-read
	Running conversion: csr-coo
DEBUG: begin csr-coo
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin components::convert_idxs_to_ptrs
DEBUG: end   components::convert_idxs_to_ptrs
DEBUG: begin repetition
DEBUG: begin copy(<typename>)
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin components::convert_ptrs_to_idxs
DEBUG: end   components::convert_ptrs_to_idxs
DEBUG: end   copy(<typename>)
DEBUG: end   repetition
DEBUG: end   csr-coo
DEBUG: end   stencil(100, 7pt)
